server/controllers/userControllers.js

```diff
@@ -15,39 +15,63 @@ class userControllers {
       phoneNumber,
       username,
     } = req.body;
-    const newUser = [
-      {
-        id: Users.length + 1,
-        firstname,
-        lastname,
-        email,
-        password,
-        phoneNumber,
-        username,
-        isAdmin: false,
-      },
-    ];
+    const newUser = {
+      firstname,
+      lastname,
+      email,
+      password,
+      phoneNumber,
+      username,
+      isAdmin: false,
+    };
+
     const hashedPassword = hashPassword(password);
     const userToken = signJwt(newUser);
     newUser.password = hashedPassword;
-    if (email === newUser.email) {
+    const userEmail = newUser.email;
+    if (email === userEmail) {
       Users.push(newUser);
       return res.status(201).json({
         status: res.statusCode,
         message: 'user Created successfull',
         data: {
           userToken,
+          id: Users.length + 1,
           firstname,
           lastname,
           email,
           phoneNumber,
           username,
+          isAdmin: false,
         },
       });
     }
     return res.status(400).json({
       status: res.statusCode,
-      message: 'email doesn\'t match',
+      message: "email doesn't match",
```

Comment:
@mic7x4 email doesn't match with what?

Reply:
Email given or Entered doesn't match with the one we have in our []